go/mysql/server.go

```diff
@@ -315,15 +306,16 @@ func NewListenerWithConfig(cfg ListenerConfig) (*Listener, error) {
 		authServer:          cfg.AuthServer,
 		handler:             cfg.Handler,
 		listener:            l,
-		ServerVersion:       cfg.MySQLServerVersion,
+		ServerVersion:       cfg.Handler.Env().MySQLVersion(),
```

Comment:
Can Handler and/or Env() be nil here? IMO it's worth a safety check here that returns an error or even an explicit panic with a message if it really shouldn't happen and we can't proceed. Same for the `cfg.Handler` call chains below.

Reply:
It should never ever be nil under any circumstance. That it panics in that case is good. I  don't think we should guard this, as it means many guards all over the place then (the env is used in many other places too).